### PR TITLE
Wip write path optimization

### PR DESCRIPTION
**Pull request by somnathr**
This is the pull request for the FileStore write path optimization work.

### CODE REVIEW

**Comment by yuyuyu101** [comment]
From my quick view, it looks like discarding most of throttling and make async op inline to reduce op latency. On the other hand, it make sync thread high frequency, so we can avoid potential WBThrottle bottleneck.
The only concern is the logic between FileJournal and SyncEntry, It makes write path much more complicated.

File: src/os/FileStore.cc [hunk 1]
```diff
@@ -3567,169 +3609,227 @@ class SyncEntryTimeout : public Context {
...
+    bool do_commit = false;
+    utime_t start_commit = ceph_clock_now(g_ceph_context);
+    while (!stop) {
+      lock.Unlock();
+      uint64_t seq_to_commit = 0;
+      if (do_commit) {
+        start_commit = ceph_clock_now(g_ceph_context);
+        do_commit = false;
+      }
+      applied_seq_lock.Lock();
```
Comment (yuyuyu101) [inline src/os/FileStore.cc]: Is this lock will be bottleneck?

Comment (somnathr) [inline src/os/FileStore.cc]: Not seeing this as bottleneck in my testing so far..

**Comment by yuyuyu101** [comment]
Could you show some numbers about how much we can improved based on your env

**Comment by somnathr** [comment]
Loic,
Not sure why make check failed initially..I saw test/ceph_objectstore_tool.py failed..But, unable to reproduce in my setup. I reran

**Comment by somnathr** [comment]
Sorry, closed it accidentally..

**Comment by somnathr** [comment]
Please see this for the benchmark result..
https://docs.google.com/presentation/d/15-Uqk0b4s1fVV1cG1G6Kba9xafcnIoLvfq8LUY7KBL0/edit#slide=id.p4

**Comment by somnathr** [comment]
Please refer to https://github.com/ceph/ceph/pull/6123

**Comment by dachary** [comment]
@somnathr the ceph_objecstore_tool make check error is a false negative, you can ignore it. It was fixed yesterday ( http://tracker.ceph.com/issues/13176 ) and you should not see it again :-)

**Pull request closed without merging**